host_core/lib/host_core/host.ex

```diff
@@ -47,7 +47,48 @@ defmodule HostCore.Host do
      %State{
        host_key: opts[:host_key],
        lattice_prefix: opts[:lattice_prefix]
-     }}
+     }, {:continue, :query_lattice_cache}}
+  end
+
+  # TODO: Evaluate downsides of a potential race condition here. May be better to simply execute this as part of the init function
+  @impl true
+  def handle_continue(:query_lattice_cache, state) do
+    Logger.info("Querying lattice for cache...")
+
+    results =
+      ParallelTask.new()
```

Comment:
I <3 this paralleltask

Reply:
It's a pretty sweet package

Reply:
And it's really nice here because we have a few queries that are okay to timeout, but we don't necessarily want to wait sequentially for them to timeout